modules/citrus-http/src/main/java/com/consol/citrus/http/message/HttpMessage.java

```diff
@@ -195,9 +198,11 @@ public HttpMessage queryParams(String queryParamString) {
         header(HttpMessageHeaders.HTTP_QUERY_PARAMS, queryParamString);
         header(DynamicEndpointUriResolver.QUERY_PARAM_HEADER_NAME, queryParamString);
 
-        this.queryParams = Stream.of(queryParamString.split(",")).map(keyValue -> Optional.ofNullable(StringUtils.split(keyValue, "=")).orElse(new String[] {keyValue, ""}))
-                                                                        .filter(keyValue -> StringUtils.hasText(keyValue[0]))
-                                                                        .collect(Collectors.toMap(keyValue -> keyValue[0], keyValue -> keyValue[1]));
+        this.queryParams = Stream.of(queryParamString.split(","))
+                .map(keyValue -> Optional.ofNullable(StringUtils.split(keyValue, "=")).orElse(new String[] {keyValue, ""}))
+                .filter(keyValue -> StringUtils.hasText(keyValue[0]))
+                .collect(Multimaps.toMultimap(keyValue -> keyValue[0], keyValue -> keyValue[1], HashMultimap::create));
```

Comment:
We found out that the `Multimaps.toMultimap` collector is experimental in the current guava version. If we want to keep the Multimap in the project, we should make sure that the collector is stable in the newest guava version. Otherwise, we should switch back to a classical `Map<String, Collection<String>> implementation internally as well.